File: front-end/src/app/entitys/produit.ts
export class produit {
    id_produit: number |null;
    lib_produit: string;
  prix_produit: number;
  description_produit: string;
  date_ajoute: Date;
  prix_liv: number;
    laboratoire: any ;
    famille: any;
 constructor(id_produit: number |null,lib_produit: string,prix_produit: number,description_produit: string,date_ajoute: Date,prix_liv: number,laboratoire: any,famille: any){
    this.id_produit=id_produit;
    this.lib_produit=lib_produit;
    this.prix_produit=prix_produit;
    this.description_produit=description_produit;
    this.date_ajoute=date_ajoute;
    this.prix_liv=prix_liv;
    this.laboratoire=laboratoire;
    this.famille=famille;
 }
}

File: Back-end/src/main/java/com/example/projetJEE/entitys/produit.java
package com.example.projetJEE.entitys;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Date;

@Entity
@Data
public class produit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id_produit;
    private String lib_produit;
    private String description_produit;
    private double prix_produit;
    private Date date_ajoute;
    private float prix_liv;

    @ManyToMany(mappedBy = "produits")
    @JsonIgnore
    private Collection<user> users = new ArrayList<>();

    @ManyToMany(mappedBy = "produits")
    @JsonIgnore
    private Collection<facture> factures = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "id_laboratoire")
    private laboratoire laboratoire;

    @ManyToOne
    @JoinColumn(name = "id_famille")
    private famille famille;
}
